src/metaswitch/clearwater/config_manager/etcd_synchronizer.py

```diff
@@ -154,3 +122,50 @@ def read_from_etcd(self):
             # function again after we return, causing the read to be retried.
 
         return value
+
+    def waitHelper(self, timeout, waitIndex):
+        result = None
+        # Calculate the args for the wait
+        args = dict(wait=True,
+                    timeout=timeout,
+                    recursive=False,
+                    quorum=True)
+        if waitIndex is not None:
+            args['waitIndex'] = result.modifiedIndex + 1
+
+        try:
+            # Wait for the key to change
+            result = self._client.read(full_key,
+                                       **args)
+            break
+        except urllib3.exceptions.TimeoutError:
+            # Timeouts after 5 seconds are expected, so ignore them
+            # - unless we're terminating, we'll stay in the while
+            # loop and try again
+            pass
+        except etcd.EtcdEventIndexCleared as e:
```

Comment:
Don't need to capture `e` as we never reference it, don't know if this has any effect though.